contracts/core/libraries/QFHelper.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+/// @title QF Helper Library
+/// @notice A helper library for Quadratic Funding
+/// @dev Handles the donation and voting of recipients and calculates the matching amount
+///      for each recipient using the Quadratic Funding formula
+library QFHelper {
+    /// Using EnumerableSet for EnumerableSet.AddressSet to store the recipients
+    using EnumerableSet for EnumerableSet.AddressSet;
+    /// Using SafeMath for uint256 calculations
+    using SafeMath for uint256;
+
+    /// @notice Error thrown when the number of recipients and amounts are not equal
+    error QFHelper_LengthMissmatch();
+
+    /// Struct that defines a donation
+    /// @param amount The amount of the donation
+    /// @param funder The address of the funder
+    struct Donation {
+        uint256 amount;
+        address funder;
+    }
+
+    /// Struct that defines the state of the donations to recipients
+    /// @param recipients The set of recipients
+    /// @param donations The donations for each recipient
+    struct State {
+        EnumerableSet.AddressSet recipients;
+        mapping(address => Donation[]) donations;
+    }
+
+    /// @notice Calculate the square root of a number (Babylonian method)
+    /// @param x The number
+    /// @return y The square root
+    function _sqrt(uint256 x) internal pure returns (uint256 y) {
```

Comment:
To be consistent with other functions in the file parameters should be `_x` and `_y`.